File: ios/Kwift/com/lightningkite/kwift/shared/DateFormattingExtras.swift
//Package: com.lightningkite.kwift.shared
//Converted using Kwift2

import Foundation



extension DateAlone {
    public func format(clockPartSize: ClockPartSize) -> String {
        return dateFrom(self, TimeAlone(0, 0, 0)).format(clockPartSize, ClockPartSize.None)
    }
    public func format(_ clockPartSize: ClockPartSize) -> String {
        return format(clockPartSize: clockPartSize)
    }
}
 

extension TimeAlone {
    public func format(clockPartSize: ClockPartSize) -> String {
        return dateFrom(Date().dateAlone, self).format(ClockPartSize.None, clockPartSize)
    }
    public func format(_ clockPartSize: ClockPartSize) -> String {
        return format(clockPartSize: clockPartSize)
    }
}
 

File: android/src/main/java/com/lightningkite/kwift/shared/DateFormattingExtras.kt
package com.lightningkite.kwift.shared

import com.lightningkite.kwift.actual.*
import java.util.*

fun DateAlone.format(clockPartSize: ClockPartSize): String
        = dateFrom(this, TimeAlone(0, 0, 0)).format(clockPartSize, ClockPartSize.None)
fun TimeAlone.format(clockPartSize: ClockPartSize): String
        = dateFrom(Date().dateAlone, this).format(ClockPartSize.None, clockPartSize)
